diff -Nru abuse_sdl-0.6.1/src/sdlport/setup.cpp abuse_sdl-0.6.1.new/src/sdlport/setup.cpp
--- abuse_sdl-0.6.1/src/sdlport/setup.cpp	Tue Feb  5 09:57:49 2002
+++ abuse_sdl-0.6.1.new/src/sdlport/setup.cpp	Sat Mar 22 14:28:47 2003
@@ -301,12 +301,20 @@
 	// Set the savegame directory
 	char *homedir;
 	char *savedir;
+	char *config_dir = getenv("CONFIG_DIR");
 	FILE *fd = NULL;
 
-	if( (homedir = getenv( "HOME" )) != NULL )
+	if (config_dir)
+	{
+		savedir = (char *)jmalloc( strlen( config_dir ) + 7, "savedir" );
+		sprintf( savedir, "%s/abuse/\0", config_dir );
+	} else if( (homedir = getenv( "HOME" )) != NULL )
 	{
 		savedir = (char *)jmalloc( strlen( homedir ) + 8, "savedir" );
 		sprintf( savedir, "%s/.abuse/\0", homedir );
+	}
+	if (config_dir || homedir)
+	{
 		// Check if we already have a savegame directory
 		if( (fd = fopen( savedir, "r" )) == NULL )
 		{
